include/nn/crypto.h

```diff
@@ -13,10 +13,119 @@ void GenerateSha256Hash(void*, ulong, void const*, ulong);
 
 class Sha256Context;
 
-void DecryptAes128Cbc(void*, u64, void const*, u64, void const*, u64, void const*, u64);
-void EncryptAes128Cbc(void*, u64, void const*, u64, void const*, u64, void const*, u64);
-void DecryptAes128Ccm(void*, u64, void*, u64, void const*, u64, void const*, u64, void const*, u64,
-                      void const*, u64, u64);
+// Util Functions
+bool IsSameBytes(const void* data1, const void* data2, u64 size);
+Result GenerateCryptographicallyRandomBytes(void* dstBuffer, u64 dstBufferSize); // dstBufferSize max size is 0x38
+
+// MD5 Hash
+void GenerateMd5Hash(void* dstBuffer, u64 dstBufferSize, const void* srcBuffer, u64 srcBufferSize);
+
+// SHA-1 Hash
+void GenerateSha1Hash(void* dstBuffer, u64 dstBufferSize, const void* srcBuffer, u64 srcBufferSize);
+
+// SHA-256 Hash
+void GenerateSha256Hash(void* dstBuffer, u64 dstBufferSize, const void* srcBuffer,
+                        u64 srcBufferSize);
+
+// SHA-512 Hash
+void GenerateSha512Hash(void* dstBuffer, u64 dstBufferSize, const void* srcBuffer,
+                        u64 srcBufferSize);
+
+// HMAC-SHA1
+void GenerateHmacSha1Mac(void* dstBuffer, u64 dstBufferSize, const void* srcBuffer,
+                         u64 srcBufferSize, const void* hmacKey, u64 hmacKeySize);
+
+// HMAC-SHA256
+void GenerateHmacSha256Mac(void* dstBuffer, u64 dstBufferSize, const void* srcBuffer,
+                           u64 srcBufferSize, const void* hmacKey, u64 hmacKeySize);
+
+// AES-128 CBC Decrypt
+u64 DecryptAes128Cbc(void* dstBuffer, u64 dstBufferSize, const void* key, u64 keySize,
+                     const void* iv, u64 ivSize, const void* srcBuffer,
+                     u64 srcBufferSize);  // returns srcBufferSize
```

Comment:
The comment should go above the function declaration so that it looks nicer once formatted with clang-format